### ISSUE TITLE
Error after three (reset) command

### ISSUE THREAD

**Issue by sy6sy2** [CVC4/CVC4#4866]
**Describe the bug**
After three `(reset)` command CVC4 (i) ignore `--bv-print-consts-as-indexed-symbols` option and (ii) trigger an error on `(reset)` command.

Reproduce the bug with this simple formula and the command given under the formula:

```SMT
(set-option :produce-models true)
(set-logic QF_ABV)

(declare-fun a () (_ BitVec 32))
(assert (= a (_ bv1 32)))

(check-sat)
(get-value (a))

(reset)

(set-option :produce-models true)
(set-logic QF_ABV)

(declare-fun a () (_ BitVec 32))
(assert (= a (_ bv1 32)))

(check-sat)
(get-value (a))

(reset)

(set-option :produce-models true)
(set-logic QF_ABV)

(declare-fun a () (_ BitVec 32))
(assert (= a (_ bv1 32)))

(check-sat)
(get-value (a))

(reset)

(set-option :produce-models true)
(set-logic QF_ABV)

(declare-fun a () (_ BitVec 32))
(assert (= a (_ bv1 32)))

(check-sat)
(get-value (a))
```

Result:

```bash
sat
((a (_ bv1 32)))
sat
((a (_ bv1 32)))
sat
((a #b00000000000000000000000000000001))
(error "Parse Error: /tmp/test.smt2:32.6: expected SMT-LIBv2 command, got `reset'.

  (reset)
   ^
")
```

*Note*: If the bug is triggered with an SMT2 input file please minimize it with [ddsmt](https://github.com/aniemetz/ddSMT).

**Command line arguments**: `cvc4 --bv-print-consts-as-indexed-symbols /tmp/test.smt2`
**CVC4 version/commit**: This is CVC4 version 1.9-prerelease [git master 33b96c65]
compiled with GCC version 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.62)
on Aug 10 2020 11:27:09
**Operating system**: macOS 10.15.6

**`configure.sh` options**
Paste `configure.sh` here.

**`configure.sh` output**
Paste the output of `configure.sh` here.

**Comment by nafur** [comment]
Indeed, this error is triggered by
```
(reset)
(reset)
(reset)
```

**Comment by nafur** [comment]
After many detours, I found the following:
- The parser error is thrown at https://github.com/CVC4/CVC4/blob/master/src/parser/smt2/Smt2.g#L537 because the [`smt25Command` rule](https://github.com/CVC4/CVC4/blob/master/src/parser/smt2/Smt2.g#L512) did not match. We would expect the [`RESET_TOK`](https://github.com/CVC4/CVC4/blob/master/src/parser/smt2/Smt2.g#L806) to match, but it doesn't.
- Actually, the lexer does not produce a `RESET_TOK` token but a `SIMPLE_SYMBOL`. We can find that the [`RESET_TOK` lexer rule](https://github.com/CVC4/CVC4/blob/master/src/parser/smt2/Smt2.g#L2251) is conditional on `PARSER_STATE->v2_5()`.
- [`v2_5()`](https://github.com/CVC4/CVC4/blob/master/src/parser/smt2/smt2.h#L281) simply checks `d_solver->getOptions().getInputLanguage()` and we thus conclude that somehow the `inputLanguage` option changes. However, `::set()` is not called on this option after the initial setup.
- Now we look at that we actually do on [`reset`](https://github.com/CVC4/CVC4/blob/master/src/smt/smt_engine.cpp#L2096): we save `d_originalOptions` and reconstruct the `SmtEngine` with these. However, `d_originalOptions` is not set in the constructor but only when we call `SmtEngine::notifyStartParsing()`.

TL;DR: the first `reset` clears `d_originalOptions`, the second `reset` clears `d_options`, and from thereon we have `LANG_AUTO` which does not support `(reset)`.

Thanks for the issue, I'll add a fix...

**Comment by sy6sy2** [comment]
Thank you for taking the time to address this issue!
